flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KillerWatchDog.java

```diff
@@ -42,12 +42,12 @@
 	@SuppressWarnings("deprecation")
 	@Override
 	public void run() {
-		final long deadline = System.currentTimeMillis() + timeout;
+		final long deadline = System.nanoTime() + timeout * 1_000_000L;
 		long now;
 
-		while (toKill.isAlive() && (now = System.currentTimeMillis()) < deadline) {
+		while (toKill.isAlive() && (now = System.nanoTime()) < deadline) {
 			try {
-				toKill.join(deadline - now);
+				toKill.join((deadline - now) / 1_000_000L);
```

Comment:
I think the fix is fine. The rounding will result in the originally used millisecond precision anyway.